erts/emulator/hipe/hipe_arm.c

```diff
@@ -31,24 +31,39 @@
 #include "hipe_native_bif.h"	/* nbif_callemu() */
 #include "hipe_bif0.h"
 
+#ifndef __has_builtin
+# define __has_builtin(x) 0
+#endif
+
 /* Flush dcache and invalidate icache for a range of addresses. */
 void hipe_flush_icache_range(void *address, unsigned int nbytes)
 {
-#if defined(__ARM_EABI__)
+    void* end = (char*)address + nbytes;
+
+#if ERTS_AT_LEAST_GCC_VSN__(4, 3, 0) || __has_builtin(__builtin___clear_cache)
+    __builtin___clear_cache(address, end);
+#elif defined(__clang__) && defined(__FreeBSD__)
```

Comment:
Does the clang code really need the FreeBSD restriction?  One would assume its builtins to be OS-agnostic (even if they have to be OS-specific internally).